src/cmd/wallet.rs

```diff
@@ -125,6 +130,16 @@ fn list_keychain_addresses(
                     .map(|utxos| Amount::from_sat(utxos.iter().map(|utxo| utxo.txout.value).sum()))
                     .unwrap_or(Amount::ZERO);
 
+                let address_used = txns
+                    .iter()
+                    .flat_map(|tx_details| tx_details.transaction.as_ref())
+                    .flat_map(|tx| tx.output.iter())
+                    .any(|o| &o.script_pubkey == script);
```

Comment:
I think use this to build a hashmap of script_pubkey to vec of transactions first. Then you can just lookup each address as you iterate over them.